lighthouse-core/config/config.js

```diff
@@ -269,7 +275,16 @@ class Config {
     }
 
     // We don't want to mutate the original config object
-    configJSON = JSON.parse(JSON.stringify(configJSON));
+    let inputConfig = configJSON;
+    configJSON = JSON.parse(JSON.stringify(inputConfig));
+    // Copy arrays that could contain plugins to allow for programmatic
+    // injection of plugins.
```

Comment:
I agree. You rightly identified that (for this PR) I just implemented the shortest path to making this work. I figure someone more familiar with Lighthouse than me might want design input on some kind of conditional-cloner.